lib/wasix/src/journal/effector/syscalls/path_create_directory.rs

```diff
@@ -20,14 +26,18 @@ impl JournalEffector {
         fd: Fd,
         path: &str,
     ) -> anyhow::Result<()> {
-        crate::syscalls::path_create_directory_internal(ctx, fd, path).map_err(|err| {
-            anyhow::format_err!(
-                "journal restore error: failed to create directory path (fd={}, path={}) - {}",
-                fd,
-                path,
-                err
-            )
-        })?;
+        if fd == VIRTUAL_ROOT_FD {
```

Comment:
Why is this necessary?
I assume for restoration?

Should have a comment explaining the need.

Reply:
How does the restoration interact with mounting?

Reply:
Added this comment

``` rust
/// Used for interacting with the file system when it has no
/// pre-opened file descriptors at the root level. Normally
/// a WASM process will do this in the libc initialization stage
/// however that does not happen when the WASM process has never
/// been run. Further that logic could change at any time in libc
/// which would then break functionality. Instead we use this fixed
/// file descriptor
///
/// This is especially important for fuse mounting journals which
/// use the same syscalls as a normal WASI application but do not
/// run the libc initialization logic
```